core/meta/src/TSchemaRule.cxx

```diff
@@ -913,14 +914,23 @@ Bool_t TSchemaRule::ProcessChecksum( const TString& checksum ) const
    // Check the validity of each list element
    /////////////////////////////////////////////////////////////////////////////
 
-   std::list<std::string>::iterator it;
-   for( it = checksums.begin(); it != checksums.end(); ++it ) {
-      if( !Internal::TSchemaRuleProcessor::IsANumber( *it ) ) {
+   for(const auto& checksumStr : checksums) {
+      std::istringstream converter(checksumStr);
```

Comment:
Can you put the parsing into a separate function (we might use it somewhere else at somepoint)?

Reply:
Sure, no problem